finagle-core/src/main/scala/com/twitter/finagle/tracing/Id.scala

```diff
@@ -84,26 +120,31 @@ object TraceId {
         traceId.flags.setFlag(Flags.SamplingKnown)
     }
 
-    val bytes = new Array[Byte](32)
+    // For backward compatibility for TraceID: 40 bytes if 128bit, 32 bytes if 64bit
+    val bytes = new Array[Byte](if(traceId.traceIdHigh.isDefined) 40 else 32)
     ByteArrays.put64be(bytes, 0, traceId.spanId.toLong)
     ByteArrays.put64be(bytes, 8, traceId.parentId.toLong)
     ByteArrays.put64be(bytes, 16, traceId.traceId.toLong)
     ByteArrays.put64be(bytes, 24, flags.toLong)
+    if(traceId.traceIdHigh.isDefined) ByteArrays.put64be(bytes, 32, traceId.traceIdHigh.get.toLong)
     bytes
   }
 
   /**
    * Deserialize a TraceId from an array of bytes.
+   * Allows for 64-bit or 128-bit trace identifiers.
    */
   def deserialize(bytes: Array[Byte]): Try[TraceId] = {
-    if (bytes.length != 32) {
-      Throw(new IllegalArgumentException("Expected 32 bytes"))
+    if (bytes.length != 32 && bytes.length != 40) {
+      Throw(new IllegalArgumentException("Expected 32 or 40 bytes"))
```

Comment:
can you include `bytes.length` in the failure? it can be helpful for debugging. something like:

`Throw(new IllegalArgumentException("Expected 32 or 40 bytes, was: " + bytes.length))`